### PR TITLE
Feature/codecov

### PR DESCRIPTION
**Pull request by lebdron**
**Previous PR did not seem to work**

## What is this pull request?

This PR makes to integrate codecov.io that comments on coverage reporting every creating a pull request into this repository.
   
## Why do you implement it? Why do we need this pull request?
For quality assurance.  

## Additional dependencies
- codecov.io
- lcov (for generating a coverage reporting)

## P.S.

We need to add the CODECOV_TOKEN in CircleCI environment variable list. 
The token can be obtained from the following.  -> https://codecov.io/gh/hyperledger/iroha

### CODE REVIEW

**Comment by neewy** [comment]
@a-r-g-v 
sorry, it seems to be fixed now :)

**Comment by lebdron** [comment]
https://codecov.io/gh/hyperledger/iroha/tree/561fb4d7f68c1ba527313036fab7c7f1893e3111/irohad/validation/impl
does not include stateful validator
https://github.com/hyperledger/iroha/tree/feature/codecov/irohad/validation/impl

**Comment by neewy** [comment]
@a-r-g-v https://codecov.io/gh/hyperledger/iroha/tree/9bcfa2c9403befb32c8a2e2507c1264265782fe0/irohad/validation/impl still some files are missed. Can you check lcov report so that we can understand where the problem is?

File: .circleci/config.yml [hunk 1]
```diff
@@ -37,9 +44,32 @@ jobs:
...
+          name: reporting coverage
+          command : |
+            lcov --capture --directory $IROHA_BUILD --output-file $IROHA_COVERAGE
+            IROHA_COVERAGE_TARGET_DIRECTORIES=('irohad' 'iroha-cli' 'libs')
+            for directory in ${IROHA_COVERAGE_TARGET_DIRECTORIES[@]}; do
+              lcov --extract $IROHA_COVERAGE "$IROHA_HOME/$directory/*" > ${IROHA_COVERAGE}_$directory
+            done
+
+            BASE_COVERAGE_REPORT=${IROHA_COVERAGE_TARGET_DIRECTORIES[0]}
+            IROHA_COVERAGE_TARGET_DIRECTORIES=(${IROHA_COVERAGE_TARGET_DIRECTORIES[@]:1})
```
Comment (l4l) [inline .circleci/config.yml]: Why "${IROHA_COVERAGE_TARGET_DIRECTORIES[@]**:1**}" ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -37,9 +44,32 @@ jobs:
...
+            for directory in ${IROHA_COVERAGE_TARGET_DIRECTORIES[@]}; do
+              lcov --add-tracefile ${IROHA_COVERAGE}_$directory --add-tracefile ${IROHA_COVERAGE}_$BASE_COVERAGE_REPORT > $IROHA_COVERAGE
```
Comment (l4l) [inline .circleci/config.yml]: It seems that **IROHA_COVERAGE** is same for all cycles of the loop. Is it the right behavior?

File: .circleci/config.yml (hunk 1 above)
Comment (l4l) [inline .circleci/config.yml]: Why "${IROHA_COVERAGE_TARGET_DIRECTORIES[@]**:1**}" ?

File: .circleci/config.yml (hunk 2 above)
Comment (l4l) [inline .circleci/config.yml]: It seems that **IROHA_COVERAGE** is same for all cycles of the loop. Is it the right behavior, it seems they're overwriting each other?

**Comment by lebdron** [comment]
https://github.com/hyperledger/iroha/pull/559

**Pull request closed without merging**